### PR TITLE
Fix: Midea heat pump outside temperature issue when below zero

### PR DESCRIPTION
**Pull request by erikgieseler**
Fix: Midea heat pump outside temperature issue when below zero https://github.com/georgezhao2010/midea_ac_lan/issues/427

### CODE REVIEW

File: custom_components/midea_ac_lan/midea/devices/c3/message.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def __init__(self, body, data_offset=0):
             + (body[data_offset + 7] << 8)
             + (body[data_offset + 8])
         )
-        self.outdoor_temperature = int(body[data_offset + 9])
+        self.outdoor_temperature = body[data_offset + 9] - 256 if body[data_offset + 9] > 127 else body[data_offset + 9]
```
Comment (chemelli74) [inline custom_components/midea_ac_lan/midea/devices/c3/message.py:171]:
```suggestion
        base_value = body[data_offset + 9]
        self.outdoor_temperature = (base_value - 256) if base_value > 127 else base_value
```

Comment (erikgieseler) [reply custom_components/midea_ac_lan/midea/devices/c3/message.py:171]: I tested the change with my own heat pump and it works.

**Review by rokam — APPROVED** [review]
LGTM. Although there is a linting error to be fixed.

**Comment by wuwentao** [comment]
@erikgieseler 
you may need to fix the github action error, it show the solution as below:

```
  -        self.outdoor_temperature = (base_value - 256) if base_value > 127 else base_value
  +        self.outdoor_temperature = (
  +            (base_value - 256) if base_value > 127 else base_value
  +        )
```

**Comment by wuwentao** [comment]
we can accept the changes, and port it to midea-local.

**Comment by erikgieseler** [comment]
The action errors are fixed

**Review by chemelli74 — APPROVED** [review]
LGTM. 

Just a small question: is that valid only for "c3" devices or there is something similar for others ?

**Comment by erikgieseler** [comment]
> LGTM.
> 
> Just a small question: is that valid only for "c3" devices or there is something similar for others ?

Unfortunately, I can't answer that because I only have a C3 heat pump.

> LGTM.
> 
> Just a small question: is that valid only for "c3" devices or there is something similar for others ?

Unfortunately, I can't answer that because I only have a C3 heat pump. Perhaps the same problem also exists with other devices where the exact same old code was used.

**Comment by chemelli74** [comment]
We need to rework the code and remove all those duplicates.
I'm nearly done with the typing, next job will be the clean-up.

**Comment by erikgieseler** [comment]
> We need to rework the code and remove all those duplicates. I'm nearly done with the typing, next job will be the clean-up.

I have done a search with the old code across all other devices. 
There are no duplicates that use the same code.

**Pull request merged**